Cesium3DTilesSelection/src/RasterOverlayUpsampler.cpp

```diff
@@ -85,26 +82,29 @@ CesiumAsync::Future<TileLoadResult> RasterOverlayUpsampler::loadTileContent(
   }
 
   const CesiumGltf::Model& parentModel = pParentRenderContent->model.value();
-  return asyncSystem.runInWorkerThread([&parentModel,
-                                        transform = tile.getTransform(),
-                                        textureCoordinateIndex = index,
-                                        TileID = *pTileID]() mutable {
-    auto model = upsampleGltfForRasterOverlays(
-        parentModel,
-        TileID,
-        textureCoordinateIndex);
+  return loadInput.asyncSystem.runInWorkerThread(
+      [&parentModel,
+       transform = loadInput.tile.getTransform(),
+       textureCoordinateIndex = index,
+       TileID = *pTileID]() mutable {
+        auto model = upsampleGltfForRasterOverlays(
+            parentModel,
+            TileID,
+            textureCoordinateIndex);
 
-    return TileLoadResult{
-        TileRenderContent{std::move(model)},
-        std::nullopt,
-        std::nullopt,
-        TileLoadResultState::Success,
-        nullptr,
-        {}};
-  });
+        return TileLoadResult{
+            TileRenderContent{std::move(model)},
+            std::nullopt,
+            std::nullopt,
+            std::nullopt,
+            nullptr,
+            {},
+            TileLoadResultState::Success};
+      });
 }
 
-bool RasterOverlayUpsampler::updateTileContent([[maybe_unused]] Tile& tile) {
-  return false;
+TileChildrenResult
+RasterOverlayUpsampler::createTileChildren([[maybe_unused]] const Tile& tile) {
+  return {{}, TileLoadResultState::Success};
```

Comment:
the code for subdividing tile for upsampling in `TilesetContentManager` can be put in here. The only concern I have is `createQuadtreeSubdividedChildren()` is called when `RasterMappedTo3DTiles::update()` indicates if more raster is available. So when moving `createQuadtreeSubdividedChildren()` to `TilesetContentLoader::createTileChildren()`, it will be called before `RasterMappedTo3DTiles::update()`. There is `RasterMappedTo3DTiles::isMoreDetailAvailable()`, can I use that to determine if we should create more children instead of using the update method?

Reply:
Yes, I believe that should work just fine. Just by aware that if `isMoreDetailAvailable` returns false, that doesn't necessarily mean "no". It can also mean "I don't know yet."